### PR TITLE
Migrate subscription page

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Migrate subscription page

[stage-19]

## Motivation and Context
Angular migration epic.

## How Has This Been Tested?
Tested locally.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Will update tests separately and add unimplemented pipe.

@ezequielc please review. Thanks!

File: src/app/billing/components/subscription/subscription.component.html [hunk 1]
```diff
@@ -63,27 +63,27 @@ <h1 class="mb-4">${{subscriptionFactory.getItemSubscription().plan_amount / 100
       <div class="border-container subscription-card">
         <div class="panel-body">
           <label class="mb-3">Payment Method</label>
-          <div class="flex-row madero-radio align-left mb-3" ng-repeat="card in paymentSourcesFactory.existingCreditCards">
+          <div class="flex-row madero-radio align-left mb-3" *ngFor="let card of paymentSourcesFactory.existingCreditCards">
             <div class="row-entry">
-              <input type="radio" ng-click="subscriptionFactory.changePaymentMethod($event, card)" ng-model="subscriptionFactory.item.paymentSourceId" ng-value="card.payment_source.id" name="{{card.payment_source.id}}" id="{{card.payment_source.id}}" tabindex="1">
+              <input type="radio" (click)="subscriptionFactory.changePaymentMethod($event, card)" [ngModel]="subscriptionFactory.item.paymentSourceId" [value]="card.payment_source.id" name="paymentSourceId" [id]="card.payment_source.id" tabindex="1">
```
Comment (alex-deaconu) [inline src/app/billing/components/subscription/subscription.component.html:68]: When setting an object for `ngModel`, we cannot use the `?` operator (`subscriptionFactory.item?.paymentSourceId`), or the getter `subscriptionFactory.getItemSubscription().payment_source_id`.

Because of this we have to make sure the `subscriptionFactory.item` object exists, to prevent stack errors.

See change to the factory below.

Thanks.

File: src/app/billing/components/subscription/subscription.component.html [hunk 2]
```diff
@@ -4,19 +4,19 @@
     <h1 class="app-header-title">Subscription</h1>
   </div>
 
-  <div rv-spinner rv-spinner-key="subscription-loader">
+  <div [rvSpinner]="'subscription-loader'" [rvShowSpinner]="subscriptionFactory.loading || taxExemptionFactory.loading || paymentSourcesFactory.loading">
```
Comment (ezequielc) [inline src/app/billing/components/subscription/subscription.component.html:7]: @alex-deaconu Do we need to change the format for this spinner as well? Or maybe it is done in another PR. Thanks

**Review by ezequielc — APPROVED** [review]

File: src/app/billing/components/subscription/subscription.component.html (hunk 2 above)
Comment (alex-deaconu) [reply src/app/billing/components/subscription/subscription.component.html:7]: @ezequielc next pr should address this. Thanks.

Comment (ezequielc) [reply src/app/billing/components/subscription/subscription.component.html:7]: Yep, just noticed it. Thanks! 👍
